### DISCUSSION TITLE
Hi, I have a few questions on this package.

### DISCUSSION THREAD

**Discussion by intercretasyo** [kevlatus/flutter_fortune_wheel#45] [Q&A]
First of all, thanks for this package but I have some few questions.
1. Why is it that when I spin, it is always showing the same item?
2. Is there a control on the FortuneWheel so that I can spin it only via button?
3. How can I retrieve the selected item once done in spinning.

**Comment by kevlatus** [comment]
Hi @intercretasyo, thanks for reaching out!
Without knowing the exact code you are using, I can only provide some general answers. I will base my answers on [this example](https://github.com/kevlatus/flutter_fortune_wheel/blob/main/example/lib/main.dart) of how to use this package.

> 1. Why is it that when I spin, it is always showing the same item?

It might happen that the same item is selected multiple times in a row, when using [`Fortune.randomInt()`](https://pub.dev/documentation/flutter_fortune_wheel/latest/flutter_fortune_wheel/Fortune/randomInt.html) as shown [in line 56 of the example](https://github.com/kevlatus/flutter_fortune_wheel/blob/main/example/lib/main.dart#L56).
But ultimately: when there is more than one item to choose from, the approach shown above will result in selecting different items over time.

> 2. Is there a control on the FortuneWheel so that I can spin it only via button?

There is no _control_, as in widget, provided by this package to spin using a button. As there are arbitrary ways of starting a spin, providing these kind of widgets is out of scope of this package. An example of how to start spinning with a button can be found in [the demo app](https://github.com/kevlatus/flutter_fortune_wheel/blob/0263bb6017bf2d6350c0a9cbfcc0657879cc7808/demo/lib/pages/wheel.dart#L32). 

If you want to disable touch input on the wheel, please refer to [this section of the docs](https://github.com/kevlatus/flutter_fortune_wheel#drag-behavior).

> 3. How can I retrieve the selected item once done in spinning.

The selected item is not supposed to be retrieved, since you are telling the widget, which item is selected using the `selected` stream. If you want to execute logic after the animation has ended, you may store the last emitted value somewhere and then retrieve it from inside the [`onAnimationEnd`](https://pub.dev/documentation/flutter_fortune_wheel/latest/flutter_fortune_wheel/FortuneWidget/onAnimationEnd.html) callback.

**Comment by intercretasyo** [comment]
Hi, thank you so much. Sorry for may late reply. For my 1st and 2nd question, it is now working as per my expectation. But for the 3rd, which is to get the selected value once it stops spinning, I can't get it to work. I tried to print the `selected.stream` but it is just showing `Instance of '_ControllerStream<int>'`. Can you help me on this? Thanks again.

**Reply by kevlatus** [reply]
Hey @intercretasyo,
the Dart team has some fantastic resources on streams. I believe you will find your answers by going through [this tutorial](https://dart.dev/tutorials/language/streams).

Best
Kevin

**Reply by intercretasyo** [reply]
Thank you, I have manage to make it work.

**Comment by intercretasyo** [comment]
Hi,

I am using your example here:
[https://github.com/kevlatus/flutter_fortune_wheel/blob/main/example/lib/pages/wheel.dart](url)

I saw that this code, gives me the randomize value.
`roll(Constants.fortuneValues.length)`

I have added my own variable to use as a storage.
`Widget build(BuildContext context) { `
`...   `
`    int myInt = 0;`
`...`

Then store the roll value in _**myInt**_.
`                void handleRoll() {`
`      myInt = roll(Constants.fortuneValues.length);`
`      selected.add(`
`        roll(myInt),`
`      );`
`    }`

But when I print it here, it is showing only the first roll. Succeeding roll shows the initial value.
`              onAnimationEnd: () {`
`                print('$myInt');`
`                isAnimating.value = false;`
`              },`

Please help to explain why is it that randomize value cannot be used in **_onAnimationEnd_**. Thanks!

**Reply by intercretasyo** [reply]
Ohh, I think I get it now, all the variable should be using useState also.
